pkg/controller/remotemachineset/awsactuator.go

```diff
@@ -33,16 +33,22 @@ type AWSActuator struct {
 var _ Actuator = &AWSActuator{}
 
 // NewAWSActuator is the constructor for building a AWSActuator
-func NewAWSActuator(awsCreds *corev1.Secret, region string, remoteMachineSets []machineapi.MachineSet, scheme *runtime.Scheme, logger log.FieldLogger) (*AWSActuator, error) {
+func NewAWSActuator(awsCreds *corev1.Secret, region string, pool *hivev1.MachinePool, remoteMachineSets []machineapi.MachineSet, scheme *runtime.Scheme, logger log.FieldLogger) (*AWSActuator, error) {
 	awsClient, err := awsclient.NewClientFromSecret(awsCreds, region)
 	if err != nil {
 		logger.WithError(err).Warn("failed to create AWS client")
 		return nil, err
 	}
-	amiID, err := getAWSAMIID(remoteMachineSets, scheme, logger)
-	if err != nil {
-		logger.WithError(err).Warn("failed to get AMI ID")
-		return nil, err
+	var amiID string
+	if pool.Annotations != nil && pool.Annotations[hivev1.MachinePoolImageIDOverrideAnnotation] != "" {
```

Comment:
The check for nil is not necessary.

Reply:
Then, this block can become the following.
```
amiID := pool.Annotations[hivev1.MachinePoolImageIDOverrideAnnotation]
if amiID != "" {
} else {
}
```

Reply:
For my knowledge, why is the check for nil unnecessary?

Reply:
Never mind, should have googled first: 

"Map types are reference types, like pointers or slices, and so the value of m above is nil; it doesn't point to an initialized map. A nil map behaves like an empty map when reading, but attempts to write to a nil map will cause a runtime panic; don't do that. To initialize a map, use the built in make function:"

https://blog.golang.org/maps

Reply:
Aha so it's just writes that are a problem, good to know.

Updated.